pkg/syncer/namespace/namespace_downstream_controller.go

```diff
@@ -88,32 +89,34 @@ func NewDownstreamController(
 		syncTargetWorkspace: syncTargetWorkspace,
 		syncTargetUID:       syncTargetUID,
 		syncTargetKey:       syncTargetKey,
+
+		logger: logging.WithReconciler(syncerLogger, downstreamControllerName),
```

Comment:
I don't think we want to do this. The power of a logger per context is that you get to attach information *to a particular logger* and pass that around. For instance, multi-threaded use of this common logger would not allow the controller to add the queue key being processed, the object name, etc, to the specific instance of the logger used by a particular goroutine doing a particular reconciliation. If you're holding on to a "bare" logger and creating new ones from it in every top-level case, then I'd argue why bother with the top-level one at all? When a new dev is working inside of some particular sub-method, how would they know whether to pass one through the context, or subclass one from the shared state?

Reply:
I refactored the whole according to pattern you used in other places in the code, and loggers are not stored anymore in controller structs, but passed through context when possible.